tilelang/tools/plot_layout.py

```
@@ -1,23 +1,155 @@
+import tilelang.language as T
+
+
+def _parse_formats(formats: str | list[str]) -> list[str]:
+    if isinstance(formats, str):
+        formats_str = formats.strip().lower()
+        if formats_str == "all":
+            return ["pdf", "png", "svg"]
+        if "," in formats_str:
+            return [f.strip() for f in formats_str.split(",")]
+        return [formats_str]
+    raise TypeError(f"Expected str, but got {type(formats).__name__}. Please pass a string like 'png', 'pdf', 'svg', 'all', or 'png,pdf'.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)